tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,123 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"log"
+	"os/exec"
+
+	"github.com/firepear/qsplit"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/xridge/kubestone/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
+)
+
+const (
+	iperf3SampleCR = "../../config/samples/perf_v1alpha1_iperf3.yaml"
+	e2eNamespace   = "kubestone-e2e"
+)
+
+var restClientConfig = ctrl.GetConfigOrDie()
+var client ctrlclient.Client
+var ctx = context.Background()
+var scheme = runtime.NewScheme()
+
+func init() {
+	_ = k8sscheme.AddToScheme(scheme)
+	err := perfv1alpha1.AddToScheme(scheme)
+	if err != nil {
+		panic(err)
+	}
+
+	client, err = ctrlclient.New(restClientConfig, ctrlclient.Options{Scheme: scheme})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func run(command string) (stdout, stderr string, err error) {
+	commandArray := qsplit.ToStrings([]byte(command))
+	cmd := exec.Command(commandArray[0], commandArray[1:]...)
+	var stdOut bytes.Buffer
+	var stdErr bytes.Buffer
+	cmd.Stdout, cmd.Stderr = &stdOut, &stdErr
+	err = cmd.Run()
+	if err != nil {
+		log.Printf("Error during execution of `%v`\nerr: %v\nstdout: %v\nstderr: %v\n",
+			command, err, stdOut.String(), stdErr.String())
+		return "", "", err
+	}
+
+	return stdOut.String(), stdErr.String(), nil
+}
+
+var _ = Describe("end to end test", func() {
+	Context("preparing namespace", func() {
+		_, _, err := run("kubectl create namespace " + e2eNamespace)
+		It("should succeed", func() {
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Describe("for iperf3", func() {
+		Context("creation from samples", func() {
+			_, _, err := run("kubectl create -n " + e2eNamespace + " -f " + iperf3SampleCR)
+			It("should create iperf3-sample cr", func() {
```

Comment:
minor: misleading text here, we aren't checking that the iperf3-sample cr is created or not

Reply:
i don't think it is a big problem and having trouble re-wording. I'll skip this if you don't mind.